src/averageAges.js

```diff
@@ -37,7 +44,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWomen = people.filter(({ name, sex }) => (withChildren)
+    ? sex === 'f'
+    && people.find(({ mother }) => name === mother)
+    : sex === 'f'
+  );
+
+  const womenAge = filterWomen.map(({ died, born }) => died - born);
+
+  const womenAvarageAge = womenAge.reduce((ageA, ageB) =>
+    ageA + ageB, 0) / womenAge.length;
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and on `calculateMenAverageAge`.